src/mdx2.ts

```diff
@@ -0,0 +1,141 @@
+      contents = child;
+    } else if (
+      t.isExportNamedDeclaration(child) &&
+      t.isVariableDeclaration(child.declaration) &&
+      child.declaration.declarations.length === 1
+    ) {
+      const declaration = child.declaration.declarations[0];
+      if (t.isVariableDeclarator(declaration) && t.isIdentifier(declaration.id)) {
+        const { name } = declaration.id;
+        context.namedExports[name] = declaration.init;
+      }
+    }
+  });
+  if (contents) {
+    const jsx = contents.expression as t.JSXFragment;
+    jsx.children.forEach((child) => {
+      if (t.isJSXElement(child)) {
+        if (t.isJSXIdentifier(child.openingElement.name)) {
+          const name = child.openingElement.name.name;
+          let stories;
+          if (['Canvas', 'Preview'].includes(name)) {
+            stories = genCanvasExports(child, context);
+          } else if (name === 'Story') {
+            stories = genStoryExport(child, context);
+          } else if (name === 'Meta') {
+            const meta = genMeta(child, options);
+            if (meta) {
+              if (metaExport) {
+                throw new Error('Meta can only be declared once');
+              }
+              metaExport = meta;
+            }
+          }
+          if (stories) {
+            Object.entries(stories).forEach(([key, story]) => {
+              includeStories.push(key);
+              storyExports.push(story);
+            });
+          }
+        }
+      } else if (t.isJSXExpressionContainer(child) && t.isStringLiteral(child.expression)) {
+        // Skip string literals
+      } else {
+        throw new Error(`Unexpected JSX child: ${child.type}`);
+      }
+    });
+  }
+
+  if (metaExport) {
+    if (!storyExports.length) {
+      storyExports.push('export const __page = () => { throw new Error("Docs-only story"); };');
+      storyExports.push('__page.parameters = { docsOnly: true };');
+      includeStories.push('__page');
+    }
+  } else {
+    metaExport = {};
+  }
+  metaExport.includeStories = JSON.stringify(includeStories);
+
+  const fullJsx = [
+    ...storyExports,
+    `const componentMeta = ${stringifyMeta(metaExport)};`,
+    `const mdxStoryNameToKey = ${JSON.stringify(context.storyNameToKey)};`,
+    wrapperJs,
+    'export default componentMeta;',
+  ].join('\n\n');
+
+  return fullJsx;
+}
+
+export const plugin = (store: any) => (root: any) => {
+  const estree = store.toEstree(root);
+  // toBabel mutates root, so we need to clone it
+  const clone = cloneDeep(estree);
+  const babel = toBabel(clone);
+  store.exports = extractExports(babel, {});
+
+  return root;
+};
+
+export const postprocess = (code: string, extractedExports: string) => {
+  const lines = code.toString().trim().split('\n');
+  const last = lines.pop();
+  const first = lines.shift();
+
+  return [
+    first,
+    'import { assertIsFn, AddContext } from "@storybook/addon-docs";',
+    ...lines,
```

Comment:
@shilman I believe you want this?

```suggestion
   last,
    ...lines,
```

Reply:
Great catch @jrolfs -- fixing!